### PR TITLE
Implement Data Generators in Dev

### PR DESCRIPTION
**Pull request by KitArmstrong**
Ticket: https://bcmines.atlassian.net/browse/EE-649
Changes: This adds in the data generators as part of the dev pipeline. I have set it up so that they only run on an empty collection, which will work well when we get the PR based environments set up. This was to prevent any accidental data pollution in an environment like Prod.

### CODE REVIEW

File: api/test/data_generators/generate.test.js [hunk 1]
```diff
@@ -24,7 +25,15 @@ describe('Generate Test Data', () => {
   ];
 
   describe('Generate Projects', () => {
-    test('Generator', done => {
+    test('Generator', async done => {
+    mongoose.connection.db.collection('epic').count((error, count) => {
```
Comment (KitArmstrong) [inline api/test/data_generators/generate.test.js:29]: This was done in case this script was ever somehow run over production or another DB that we didn't want to pollute. To run this script the collection must be empty.

Comment (mtCarto) [reply api/test/data_generators/generate.test.js:29]: would we also want to check if "saved" is one of the options? So that it could still update in-memory db's? Though I'm n ot sure if that's a use case we need to worry about. As I think we call the factories/generators directly in unit tests.

Comment (KitArmstrong) [reply api/test/data_generators/generate.test.js:29]: This is a good point and something we may have to add in the future. You are right that the unit tests right now call them directly and if we do use the generators they should be running over an empty collection then too. I think this would still work, but might have to adjust when we get to that point.

Comment (ActionAnalytics) [reply api/test/data_generators/generate.test.js:29]: I like the idea.  The current implementation breaks the use case of local dev adding more projects, or of a unit test starting out with a set of static projects and then firing random ones on top for edge case testing.  

A suggestion to achieve both would instead be to add an optional flag to the generator for allowing [multiple passes](https://camo.githubusercontent.com/109f88f73b4b1ebde4545c1c139bb630bc64cdf2/68747470733a2f2f732d6d656469612d63616368652d616b302e70696e696d672e636f6d2f353634782f35342f66352f36382f35346635363862653064623437363530303231363030343463653039396438392e6a7067) and have it default to not do so, so you would achieve your goal of default safety but not break existing functionality.

Here are the following areas you'd need to mod:

https://github.com/bcgov

1. /eagle-api/blob/64ff432a394eeeab91fb57f7ec5575ef30c0f495/generate.sh#L14
2. https://github.com/bcgov/eagle-api/blob/64ff432a394eeeab91fb57f7ec5575ef30c0f495/api/test/test_helper.js#L93
3. https://github.com/bcgov/eagle-api/blob/64ff432a394eeeab91fb57f7ec5575ef30c0f495/api/test/test_helper.js#L99

Comment (KitArmstrong) [reply api/test/data_generators/generate.test.js:29]: Great idea. I have updated to check an additional flag on the command.
